src/view.c

```diff
@@ -2116,8 +2117,22 @@ view_move_to_output(struct view *view, struct output *output)
 	} else if (view->maximized != VIEW_AXIS_NONE) {
 		view_apply_maximized_geometry(view);
 	} else if (view->tiled) {
+		/*
+		 * Tweak the natural geometry to also be on the new output.
+		 * This ensures that untiling keeps the view on the new
+		 * output and does not jump back to the old output.
+		 */
+		view->pending.x = output_area.x;
+		view->pending.y = output_area.y;
+		view->natural_geometry = view->pending;
```

Comment:
I can't  use `view_store_natural_geometry` (https://github.com/labwc/labwc/blob/master/src/view.c#L843) here since that function exists early for tlied windows.

Reply:
This looks strange. Won't the pending geometry have the tiled width and height at this point? We should just copy the new `x` and `y` fields into `view->natural_geometry` instead of also copying the width and height.

Reply:
Thanks, adjusted and untiling still works nicely after moving a tiled view to another output.